### ISSUE TITLE
Update docs to recommend using `init` over `constructor`

### ISSUE THREAD

**Issue by pzuraq** [ember-decorators/ember-decorators#294]
Since RFC 337 was merged, this is going to be the new normal for a while. We should update the docs to reflect this change.

**Comment by Gorzas** [comment]
@pzuraq Isn't the idea to totally remove EmberObject and use only native classes? So recommening init before constructor may counterproductive. Don't you think?

**Comment by pzuraq** [comment]
You can check out the details of [RFC 337](https://github.com/emberjs/rfcs/blob/master/text/0337-native-class-constructor-update.md) and the [Native Class Roadmap RFC](https://github.com/pzuraq/emberjs-rfcs/blob/b47e7f9ec4f02c7d27d50de64691130e7d22747d/text/0000-native-class-roadmap.md) for some more context surrounding this, but the short reason why is _injections_ and _create params_ are no longer accessible in the constructor:

```js
// invoked as <MyComponent someArg="foo" />
class MyComponent extends Component {
  @service store;
  someArg = null;

  constructor() {
    super(...arguments);
    this.store; // undefined
    this.someArg; // null
  }

  init() {
    super.init(...arguments);
    this.store; // DS.Store
    this.someArg; // "foo"
  }
}
```

Users can technically still use the `constructor`, but it is not safe as a codemod target, and these limitations could be confusing, so whenever you are extending from `EmberObject` it is recommended that you use `init`. This also means that you cannot do `new EmberObject()` or use `new` with any of its subclasses, since `create` actually triggers `init`.

However, this is _not_ the case with non-`EmberObject` based classes, since they do _not_ fire the `init` hook. In fact, the recommendation moving forward is that for all classes that do not extend from base Ember constructs, you should just use plain native classes which do not extend from `EmberObject`. 

```js
class MyClass {
  constructor(foo) {
    this.foo = foo;
  }

  @computed
  get bar() {
    return this.foo;
  }
}

let myClass = new MyClass('baz');
myClass.bar; // 'baz'
```

Base Ember constructs include:

* `Component`
* `Controller`
* `Route`
* `Helper`
* `Service`

In other words _only_ classes which are passed to the container should extend from `EmberObject` and have the `init` hook. So, you can think of `init` as "fully initialized by the container", and it's just another lifecycle hook, like `didInsertElement` or `beforeModel`.
